# Dictionary


```
my_dcit = {
  'key1': 'value1',
  'key2': 'value2',
  'key3': 'value3
}
```



In [ ]:
# value에 list 활용
dict1 = {
    'col1' : [1, 2, 3, 4],
    'col2' : [10, 20, 30, 40],
    'col3' : [100, 200, 300, 400]
}
dict1

{'col1': [1, 2, 3, 4], 'col2': [10, 20, 30, 40], 'col3': [100, 200, 300, 400]}

In [ ]:
# DataFrame으로 변경
import pandas as pd
df = pd.DataFrame(dict1)
df

col1  col2  col3
0     1    10   100
1     2    20   200
2     3    30   300
3     4    40   400

In [ ]:
type(df)

pandas.core.frame.DataFrame

# DataFrame 조사
- head(): 처음 5줄
- tail(): 마지막 5줄
- info(): 열 정보, 데이터 타입 요약
- describe(); 숫자형 열 통계 요약

In [ ]:
# seaborn으로 데이터 가져오기
# https://github.com/mwaskom/seaborn-data (해당 사이트에 들어가서 원하는 데이터를 가져올 수 있다. )
import seaborn as sns
df_iris = sns.load_dataset('iris') # https://github.com/mwaskom/seaborn-data/iris.csv 데이터 가져오기

In [ ]:
df_iris.head(10)

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa
6           4.6          3.4           1.4          0.3  setosa
7           5.0          3.4           1.5          0.2  setosa
8           4.4          2.9           1.4          0.2  setosa
9           4.9          3.1           1.5          0.1  setosa

In [ ]:
df_iris.tail(10)

sepal_length  sepal_width  petal_length  petal_width    species
140           6.7          3.1           5.6          2.4  virginica
141           6.9          3.1           5.1          2.3  virginica
142           5.8          2.7           5.1          1.9  virginica
143           6.8          3.2           5.9          2.3  virginica
144           6.7          3.3           5.7          2.5  virginica
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

In [ ]:
df_iris.info() # column명 | Non-Null Count (널값이 몇개가 있는지) | Dtype(데이터 타입은 무엇인지) float63 = 실수 & object = 문자

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [ ]:
df_iris.describe()

sepal_length  sepal_width  petal_length  petal_width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

# 표준편차(std) : 분산이 넓게 분포될수록 값이 크고, 분산이 빽빽하게 분포할 수 있도록 작은 수치가 나옴
# 25% : 상위 25% 레코드(행의 집합)
# 50% : 상위 50% 레코드(행의 집합)
# 75% : 상위 75% 레코드(행의 집합)

#  Exercise
1. titanic 데이터를 sns로 불러오고 상위 25% age 찾기

In [ ]:
import seaborn as sns
df1 = sns.load_dataset('titanic')
df1

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [ ]:
# 상위 25%를 찾으려면 describe() 함수를 사용해야함 (내가 생각한 것)
df1.describe()

survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

       survived	pclass	    age	    sibsp	    parch	    fare
25%	0.000000	2.000000	20.125000	0.000000	0.000000	7.910400
- 따라서 정답 age = 20.125000